### PR TITLE
Ngrx

### PR DESCRIPTION
**Pull request by gmq**

### CODE REVIEW

File: src/store/market/market.service.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+@Injectable()
+export class MarketService {
+  constructor(private http: HTTP) {}
+
+  public getLimitOrders(hamCode: string) {
+    return this.http
+      .get('http://pl-ham.herokuapp.com/market/limit_orders', {}, this.getHeaders(hamCode))
+      .then(
+        response => {
+          console.log(response.data);
```
Comment (monkeyci) [inline src/store/market/market.service.ts]: Calls to 'console.log' are not allowed.

File: src/store/market/market.actions.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+export const GET_LIMIT_ORDERS_FAILURE = '[Lunch] Get Limit Orders Failure';
+export const GET_LIMIT_ORDERS_SUCCESS = '[Lunch] Get Limit Orders Success';
+export const PLACE_LIMIT_ORDER = '[Lunch] Place Limit Order';
+export const PLACE_LIMIT_ORDER_FAILURE = '[Lunch] Place Limit Order Failure';
+export const PLACE_LIMIT_ORDER_SUCCESS = '[Lunch] Place Limit Order Success';
+export const PLACE_MARKET_ORDER = '[Lunch] Place Market Order';
+export const PLACE_MARKET_ORDER_FAILURE = '[Lunch] Place Market Order Failure';
+export const PLACE_MARKET_ORDER_SUCCESS = '[Lunch] Place Market Order Success';
+
+
```
Comment (monkeyci) [inline src/store/market/market.actions.ts:15]: Consecutive blank lines are forbidden

File: src/store/lunch/lunch.effects.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  );
+
+  @Effect()
+  public signUpForLunch$: Observable<Action> = this.actions$
+    .ofType(lunchActions.SIGN_UP_FOR_LUNCH)
+    .pipe(
+      withLatestFrom(this.store.select(fromRoot.getAuthHamCode)),
+      mergeMap(([_, hamCode]) =>
+        Observable.from(this.lunchService.signUpForLunch(hamCode)).pipe(
+          map(_ => new lunchActions.SignUpForLunchSuccess()),
```
Comment (monkeyci) [inline src/store/lunch/lunch.effects.ts:56]: Shadowed variable: '_'

File: src/store/lunch/lunch.actions.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+export const GET_WINNERS_FAILURE = '[Lunch] Get Winners Failure';
+export const GET_WINNERS_SUCCESS = '[Lunch] Get Winners Success';
+export const GET_CURRENT_LUNCHERS = '[Lunch] Get Current Lunchers';
+export const GET_CURRENT_LUNCHERS_FAILURE = '[Lunch] Get Current Lunchers Failure';
+export const GET_CURRENT_LUNCHERS_SUCCESS = '[Lunch] Get Current Lunchers Success';
+export const SIGN_UP_FOR_LUNCH = '[Lunch] Sign Up For Lunch';
+export const SIGN_UP_FOR_LUNCH_FAILURE = '[Lunch] Sign Up For Lunch Failure';
+export const SIGN_UP_FOR_LUNCH_SUCCESS = '[Lunch] Sign Up For Lunch Success';
+
+
```
Comment (monkeyci) [inline src/store/lunch/lunch.actions.ts:15]: Consecutive blank lines are forbidden

File: src/store/karma/karma.reducer.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    }
+    default: {
+      return state;
+    }
+  }
+}
+
+export const getKarma = (state: State) => state.karma;
+export const getErrorMessage = (state: State) => state.errorMessage;
+;
```
Comment (monkeyci) [inline src/store/karma/karma.reducer.ts]: Unnecessary semicolon

File: src/store/auth/auth.reducer.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
+/* tslint:disable typedef */
+import * as auth from './auth.actions';
+
+export interface State {
+  hamCode: string;
+  loggedIn: boolean,
```
Comment (monkeyci) [inline src/store/auth/auth.reducer.ts]: Properties should be separated by semicolons

File: src/pages/home/home.ts [hunk 7]
```diff
@@ -1,141 +1,85 @@
...
-          }
-        },
-        err => {
-          const errorMessage = JSON.parse(err.error).message;
-          this.showToast(errorMessage);
-        }
-      )
-    })
+    toast.onDidDismiss(() => {
+      this.store.dispatch(new karma.EraseError());
```
Comment (aamatte) [inline src/pages/home/home.ts]: Hmm quizás le pondría a la acción `DismissRequestError` o `DismissError`

File: src/store/karma/karma.effects.ts [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+@Injectable()
+export class KarmaEffects {
+  @Effect()
+  public login$: Observable<Action> = this.actions$.ofType(karmaActions.GET_KARMA).pipe(
+    withLatestFrom(this.store.select(fromRoot.getAuthHamCode)),
+    mergeMap(([_, hamCode]) =>
+      Observable.from(this.karmaService.getKarma(hamCode)).pipe(
+        map(karma => new karmaActions.GetKarmaSuccess(karma)),
+        catchError(err => {
+          console.error(err);
```
Comment (aamatte) [inline src/store/karma/karma.effects.ts]: podriamos matar este `console.error()`

File: src/store/auth/auth.effects.ts [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+@Injectable()
+export class AuthEffects {
+  @Effect()
+  public login$: Observable<Action> = this.actions$.ofType(authActions.LOGIN).pipe(
+    map((action: authActions.Login) => action.payload),
+    mergeMap(payload =>
+      Observable.from(this.authService.login(payload)).pipe(
+        map(_ => new authActions.LoginSuccess()),
+        catchError(err => {
+          console.error(err);
```
Comment (aamatte) [inline src/store/auth/auth.effects.ts]: Este tmbn

**Review by aamatte — COMMENTED** [review]
Los comentarios son tonteras, con eso estaría listo

**Review by aamatte — APPROVED** [review]

**Pull request merged**